src/stan/interface_callbacks/writer/stream_writer.hpp

```diff
@@ -0,0 +1,101 @@
+#ifndef STAN_INTERFACE_CALLBACKS_WRITER_STREAM_WRITER_HPP
+#define STAN_INTERFACE_CALLBACKS_WRITER_STREAM_WRITER_HPP
+
+#include <stan/interface_callbacks/writer/base_writer.hpp>
+#include <iostream>
+#include <vector>
+#include <string>
+
+namespace stan {
+  namespace interface_callbacks {
+    namespace writer {
+
+      template <class Stream>
```

Comment:
For many of the same reasons templates are used to replace class
derivations.  Here it was just to avoid relying on a specific ostream
interface (and looking up how sstream and fstream all relate to each other).

On Oct 9, 2015, at 8:54 AM, Stan buildbot <notifications@github.com> wrote:

> In src/stan/interface_callbacks/writer/stream_writer.hpp:
> 
> > @@ -0,0 +1,101 @@
> > +#ifndef STAN_INTERFACE_CALLBACKS_WRITER_STREAM_WRITER_HPP
> > +#define STAN_INTERFACE_CALLBACKS_WRITER_STREAM_WRITER_HPP
> > +
> > +#include <stan/interface_callbacks/writer/base_writer.hpp>
> > +#include <iostream>
> > +#include <vector>
> > +#include <string>
> > +
> > +namespace stan {
> > +  namespace interface_callbacks {
> > +    namespace writer {
> > +
> > +      template <class Stream>
> @betanalpha, any reason we have this templated instead of taking in an std::ostream?
> 
> —
> Reply to this email directly or view it on GitHub.
>